portfolio/src/main/java/com/google/sps/servlets/VKDataServlet.java

```diff
@@ -0,0 +1,162 @@
+            if (!country.equals("Russia") && !country.equals("")) {
+              countries.put(country, current_value == null ? value : current_value + value);
+            }
+          } catch (NumberFormatException e) {
+            System.out.println(e);
+          }
+          break;
+        
+        case "cities":
+          String city = cells[2].replace("\"", "");
+          try {
+            Integer value = Integer.valueOf(cells[4]);
+            Integer current_value = cities.get(city);
+            if (!city.equals("")) {
+              cities.put(city, current_value == null ? value : current_value + value);
+            }
+          } catch (NumberFormatException e) {
+            System.out.println(e);
+          }
+          break;
+        
+        case "gender": 
+          String gender = cells[2].replace("\"", "");
+          try {
+            Integer value = Integer.valueOf(cells[4]);
+            Integer current_value = genders.get(gender);
+            genders.put(gender, current_value == null ? value : current_value + value);
+          } catch (NumberFormatException e) {
+            System.out.println(e);
+          }
+          break;
+        
+        case "views": 
+          try {
+            Integer value = Integer.valueOf(cells[4]);
+            Map<String, Integer> views_for_year = null;
+            if (views.get(year) == null) {
+              views_for_year = new HashMap<String, Integer>();
+            } else {
+              views_for_year = views.get(year);
+            }
+            if (value != 0) {
+              Integer current_value = views_for_year.get("total");
+              views_for_year.put("total", current_value == null ? value : current_value + value);
+              views.put(year, views_for_year);
+            }
+          } catch (NumberFormatException e) {
+            System.out.println(e);
+          }
+          break;
+        case "age":
+          String age = cells[2].replace("\"", ""); 
+          try {
+            Integer value = Integer.valueOf(cells[4]);
+            Map<String, Integer> views_for_year = null;
+            if (views.get(year) == null) {
+                views_for_year = new HashMap<String, Integer>();
+            } else {
+                views_for_year = views.get(year);
+            }
+            Integer current_value = views_for_year.get(age);
+            views_for_year.put(age, current_value == null ? value : current_value + value);
+            views.put(year, views_for_year);
+          } catch (NumberFormatException e) {
+            System.out.println(e);
+          }
+          break;
+      }
+    }
+    scanner.close();
+
+    List<Map.Entry<String, Integer>> countries_list = new ArrayList<>(countries.entrySet());
+    countries_list.sort(Map.Entry.comparingByValue(Comparator.reverseOrder()));
+    for (Map.Entry<String, Integer> entry : countries_list) {
+        countries_sorted.put(entry.getKey(), entry.getValue());
+    }
```

Comment:
Maybe this is a good example to start using Java streams :) 
https://www.baeldung.com/java-hashmap-sort#value_5
There is also an example how to get the sorted linkedhashmap.

Since you are doing the same twice, please use a method instead of doubling the code.